src/pages/iou/MoneyRequestMerchantPage.js

```diff
@@ -58,7 +58,7 @@ function MoneyRequestMerchantPage({iou, route}) {
             IOU.resetMoneyRequestInfo(moneyRequestId);
         }
 
-        if (_.isEmpty(iou.participants) || (iou.amount === 0 && !iou.receiptPath) || shouldReset) {
+        if (_.isEmpty(iou.participantAccountIDs) || (iou.amount === 0 && !iou.receiptPath) || shouldReset) {
             Navigation.goBack(ROUTES.getMoneyRequestRoute(iouType, reportID), true);
         }
     }, [iou.id, iou.participants, iou.amount, iou.receiptPath, iouType, reportID]);
```

Comment:
I think you missed the dep array in this one?

```suggestion
    }, [iou.id, iou.participantAccountIDs, iou.amount, iou.receiptPath, iouType, reportID]);
```

Reply:
yeah, thanks!